CONTRIBUTING.md

```diff
@@ -0,0 +1,231 @@
+- Next to the "Edit file" button you'll find another option called "Open with github.dev" which opens a "Visual Studio Code" editor within your browser.
+
+
+## Expectations Regarding Timelines
+
+We are maintaining `synadm` in our spare time and currently are not sponsored by a company to do that. Nevertheless we try to be responsive to contributors and try to release new features as soon as they are merged.
+
+- We try to give a first review to your submitted pull-requests **within a week**, often you'll get an answer much faster though.
+- We don't expect you to finish the PR / submit the changes we suggested "in time" but would like to ask you to **communicate your personal timeline**.
+- Usually it's not critical for a feature to be finished within a time frame. If you inform us that you will be able to finish your work within a month, but not right now, it's fine, **please just let us know**!
+- If it's a serious bug we might ask you for permission to take over your branch and continue the work ourselves.
+- Unfortunately we often see contributors submitting PR's, correcting a few of our suggested changes but then disappear. Please don't do that! If you realize you won't find the time at all to continue your work, **please just let us know**. It's perfectly fine and there won't be hard feelings. If we consider your feature useful we might even take over and finish it ourselves.
+- We try to release merged features as soon as possible. Even the tiniest feature deserves a release. Sometimes we just can't make it soon enough because we don't find the time or we wait for something else because it makes sense to be released together. If a feature you submitted is merged and you'd need it urgently to be available on PyPi, **please kindly remind us** and we'll try to find the time.
+
+
+## Discuss & Design First
+
+It proved to be useful in the past if loose (feature) ideas would be discussed in #synadm:peek-a-boo.at first, which helps forming a more concrete idea that can than further be summarized into a well written github issue and finally be implemented by a `synadm` contributor or a `synadm` maintainer.
+
+- If you'd like someone else to pick up your feature idea because you are not able to code it yourself, open a detailed GitHub issue describing the feature.
+- If you'd like to code it yourself, a separate issue is not required, just describe your feature within the PR.
+
+
+## Getting the Source & Installing
+
+### Install into a Virtual Python Environment
+
+To install `synadm` into its own private virtual Python environment, FIXME
+
+<!-- omit in toc -->
+#### 1. Check Python Version
+
+`python3 --version` should show at least v3.6.x
+
+<!-- omit in toc -->
+#### 2. Clone Repo:
+
+```
+git clone https://github.com/joj0/synadm
+```
+
+<!-- omit in toc -->
+#### 3. Set up a Python3 Virtual Environment
+
+
+Create and activate a virtual environment using the python3 venv module:
+
+```
+python3 -m venv ~/.venvs/synadm
```

Comment:
Personally I wouldn't do it that way. Instead, I would do:
```
python3 -m venv venv
```
This creates a virtual environment wherever your current directory is (hopefully within the synadm folder), so when you delete a project folder, the things related to it also gets deleted.

However, because the `.gitignore` file does not ignore `venv/` but only `env`, I have to do this:
```
python3 -m venv env
```
Still located within the project, just named differently.

Second point, the virtual environment is hidden away. So when the user is trying to figure out what to delete because their drive space is close to full, they might miss that hidden folder located at home.

Reply:
It's questionable if the several options of creating and managing virtualenvs and diskspace should be included in that doc at all but I always have the feeling that people that "halfway know" about how to install a pip package later wonder why they mess up things when they install everything in there user space, even things they tinker with... That's why I tend to keep a tutorial like that in my project repos sometimes.

I personally hate when venvs are directly to be found in project repos. I want to have all of them in one place. Sometimes I just delete bunches of them when I realize that was some experiment I'll never use again. The fact that a user has problems with hidden vs. non-hidden files on a Unix system I would not consider too much or if I'm being rude I'd say: Not my problem! :-))))

I could imagine to at least make it non-hidden. Yeah good idea. And also it would not make this tiny tutorial Unix-only. (Well slash vs. backslash would still be an issue if someone would code on windows...)